src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,100 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        if (index == items.length) {
+            items = grow();
+        }
+        size += 1;
+        System.arraycopy(items, index, items, index + 1, items.length - index - 1);
```

Comment:
```suggestion
        System.arraycopy(items, index, items, index + 1, ??? );
```
let's review this part (you will need to move line 31 after that)